pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -66,8 +66,22 @@ func containsString(sli []string, str string) bool {
 	return false
 }
 
-func restartPolicyToBool(policy machinev1.GCPRestartPolicyType) *bool {
-	restart := policy == machinev1.RestartPolicyAlways
+func restartPolicyToBool(policy machinev1.GCPRestartPolicyType, preemptible bool) *bool {
+	// for more information about how the restart policy works, see the GCP docs at
+	// https://cloud.google.com/compute/docs/instances/setting-vm-host-options#settingoptions
+	restart := false
+
+	if preemptible {
+		// preemptible instances cannot be set to always restart
+		klog.V(4).Info("Setting preemptible instance restart policy to never")
```

Comment:
i added this log line because we are potentially deviating from what the user has specified in the providerSpec. it is possible that a user might have set the restart policy to "Always", which is not usable with preemptible instances. i figure we should at least log the change, although this might be something we want to add the validation webhook for gcp.